### PR TITLE
i18n(fr): update various content layer related pages

### PR DESCRIPTION
**Pull request by HiDeoo**
#### Description

This PR updates the French version of the following pages with the changes from #2707, #2676, and #2612 which are all Content Layer related to I decided to group them together in one PR:

- `guides/i18n`
- `guides/project-structure`
- `guides/site-search`
- `manual-setup`
- `reference/frontmatter`
- `reference/overrides`

### CODE REVIEW

File: docs/src/content/docs/fr/manual-setup.mdx [hunk 1]
```diff
@@ -62,20 +62,24 @@ Retrouvez toutes les options disponibles dans la [référence de configuration S
...
+import { docsLoader } from '@astrojs/starlight/loaders';
 import { docsSchema } from '@astrojs/starlight/schema';
 
 export const collections = {
-	docs: defineCollection({ schema: docsSchema() }),
+	docs: defineCollection({ loader: docsLoader(), schema: docsSchema() }),
 };
 ```
 
+Starlight supporte également [l'drapeau `legacy.collections`](https://docs.astro.build/fr/reference/legacy-flags/) où les collections sont gérées en utilisant l'implémentation de collections de contenu héritée.
```
Comment (delucis) [inline docs/src/content/docs/fr/manual-setup.mdx:80]: Is « l’drapeau » correct here? 👀 Never seen `l’` straight into a consonant before.

**Review by delucis — COMMENTED** [review]
Looks great to me; I just had one query.

File: docs/src/content/docs/fr/manual-setup.mdx (hunk 1 above)
Comment (HiDeoo) [reply docs/src/content/docs/fr/manual-setup.mdx:80]: Oh, excellent catch, that's entirely a mistake, I started with `le drapeau`, then switched to `l'option`, then switched back and switched back again after checking what translators picked on Astro Docs and I messed up the articles in the process 😆

Comment (HiDeoo) [inline docs/src/content/docs/fr/manual-setup.mdx:80]:
```suggestion
Starlight supporte également [l'option `legacy.collections`](https://docs.astro.build/fr/reference/legacy-flags/) où les collections sont gérées en utilisant l'implémentation de collections de contenu héritée.
```

**Pull request merged**